File: concrete/src/test/java/pcb/uwu/amount/finance/RentTest.java
package pcb.uwu.amount.finance;

import org.junit.Test;
import pcb.uwu.amount.base.Time;
import pcb.uwu.amount.derived.finance.InterestRate;
import pcb.uwu.unit.finance.InterestRateUnit;

import static java.math.MathContext.DECIMAL64;
import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.base.DayUnit.DAY;
import static pcb.uwu.unit.base.WeekUnit.WEEK;

public class RentTest {
	private GBP gbp = new GBP(6);
	private Time time = new Time(3, DAY);
	private InterestRate interestRate = new InterestRate(0.2, new InterestRateUnit(WEEK));

	@Test
	public void testFromMoneyAndTime() {
		assertEquals("2.00 £⋅d⁻¹", gbp.dividedBy(time, DECIMAL64).toString());
		assertEquals("1.200 £⋅wk⁻¹", gbp.multipliedBy(interestRate, DECIMAL64).toString());
	}
}

File: concrete/src/test/java/pcb/uwu/amount/finance/RentTest.kt
package pcb.uwu.amount.finance

import org.junit.Assert.assertEquals
import org.junit.Test
import pcb.uwu.amount.base.Time
import pcb.uwu.amount.derived.finance.InterestRate
import pcb.uwu.unit.base.DayUnit.DAY
import pcb.uwu.unit.base.WeekUnit.WEEK
import pcb.uwu.unit.derived.fundamental.FrequencyUnit
import java.math.MathContext.DECIMAL64

class RentTest
{
    @Test
    fun testFromMoneyAndTime()
    {
        assertEquals("2 £⋅d⁻¹",
                     GBP(6).div(Time(amount = 3, unit = DAY), DECIMAL64).toString())
        assertEquals("1.2 £⋅wk⁻¹",
                     GBP(6).multiply(InterestRate(amount = 0.2, unit = FrequencyUnit(WEEK)), DECIMAL64).toString())
    }
}